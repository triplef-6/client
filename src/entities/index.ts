export * from "./tour"
export * from "./location"
export * from "./review"
export * from "./user"
export * from "./tags"