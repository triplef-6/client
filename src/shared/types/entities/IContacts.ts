export interface IContacts {
    vk?: string
    telegram?: string
    phone: string
}