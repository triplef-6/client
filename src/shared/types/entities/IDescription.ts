export interface IDescription {
    info: string
    whatToExpect: string
    places: string[]
    topics: string[]
    orgDetails: string
    meetingPlace: string
}