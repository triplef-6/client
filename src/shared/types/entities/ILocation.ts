export interface ILocation {
    id: number
    city: string
    country: string
    tourCount: number
    region: string
    image?: string
}