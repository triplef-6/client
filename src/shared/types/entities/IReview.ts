export interface IReview {
    id: number
    name: string
    withChildren: boolean
    personCount: number
    rating: number
    positiveText: string
    negativeText: string
    tourId: number
}