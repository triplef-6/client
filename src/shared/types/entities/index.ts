export * from "./ITour.ts"
export * from "./IDescription.ts"
export * from "./ILocation.ts"
export * from "./IMe.ts"
export * from "./IContacts.ts"
export * from "./IReview.ts"
export * from "./IOrder.ts"
export * from "./IUser.ts"