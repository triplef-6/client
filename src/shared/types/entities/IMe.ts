import {IContacts, UserRole} from "@/shared/types";

export interface IMe {
    id: number
    name: string
    surname: string
    email: string
    role: UserRole
    avatar: string
    avatarFile: File | null
    tags: string[]
    rating?: number
    ratingCount?: number
    contacts?: IContacts
    info?: string
}