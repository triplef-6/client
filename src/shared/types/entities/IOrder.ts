import {ITour} from "@/shared/types";

export interface IOrder {
    id: number
    tour: ITour
    userId: number
    groupCapacity: number
}