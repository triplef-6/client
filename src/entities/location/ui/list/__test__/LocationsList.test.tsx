import { describe, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import {Locations} from "@/entities"

describe("Locations list", () => {

    it("Проверка монтирования компонента", () => {
        render(
            <MemoryRouter>
                <Locations/>
            </MemoryRouter>
        )
    })

    it("Snapshot тест", () => {
        const { container } = render(
            <MemoryRouter>
                <Locations/>
            </MemoryRouter>
        )
        expect(container).toMatchSnapshot()
    })

    it("Проверка наличия заголовка в компоненте", () => {
        render(
            <MemoryRouter>
                <Locations />
            </MemoryRouter>
        )
        expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent("Самые популярные направления")
    })

    it("Проверка наличия кнопки в компоненте", () => {
        render(
            <MemoryRouter initialEntries={["/"]}>
                <Locations />
            </MemoryRouter>
        )
        expect(screen.getByRole("button", { name: "перейти к списку городов" })).toBeInTheDocument()
    })

})