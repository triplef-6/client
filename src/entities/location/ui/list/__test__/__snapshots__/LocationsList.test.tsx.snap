// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Locations list > Snapshot тест 1`] = `
<div>
  <section
    class="_container_5ca7a1"
  >
    <h2
      class="_heading_5ca7a1"
      role="heading"
    >
      Самые популярные направления
    </h2>
    <div
      class="_list_5ca7a1"
    >
      <div
        class="_layout_802c16 _sizes_802c16 _bg_802c16 _transitions_802c16 group"
        role="locationCardBackground"
        style="background-image: url(/src/shared/assets/tempData/assets/cardBg.png);"
      >
        <div
          class="_bg_459f4f _opacity_459f4f"
        />
        <div
          class="_layout_ecf88f"
        >
          <span
            class="_country_ecf88f"
          >
            Россия
          </span>
          <span
            class="_city_ecf88f"
          >
            Омск
          </span>
          <span
            class="_count_ecf88f"
          >
            100
            + экскурсий
          </span>
        </div>
      </div>
      <div
        class="_layout_802c16 _sizes_802c16 _bg_802c16 _transitions_802c16 group"
        role="locationCardBackground"
        style="background-image: url(/src/shared/assets/tempData/assets/cardBg2.png);"
      >
        <div
          class="_bg_459f4f _opacity_459f4f"
        />
        <div
          class="_layout_ecf88f"
        >
          <span
            class="_country_ecf88f"
          >
            Россия
          </span>
          <span
            class="_city_ecf88f"
          >
            Санкт-Петербург
          </span>
          <span
            class="_count_ecf88f"
          >
            200
            + экскурсий
          </span>
        </div>
      </div>
      <div
        class="_layout_802c16 _sizes_802c16 _bg_802c16 _transitions_802c16 group"
        role="locationCardBackground"
        style="background-image: url(/src/shared/assets/tempData/assets/cardBg3.png);"
      >
        <div
          class="_bg_459f4f _opacity_459f4f"
        />
        <div
          class="_layout_ecf88f"
        >
          <span
            class="_country_ecf88f"
          >
            Россия
          </span>
          <span
            class="_city_ecf88f"
          >
            Новосибирск
          </span>
          <span
            class="_count_ecf88f"
          >
            300
            + экскурсий
          </span>
        </div>
      </div>
      <div
        class="_layout_802c16 _sizes_802c16 _bg_802c16 _transitions_802c16 group"
        role="locationCardBackground"
        style="background-image: url(/src/shared/assets/tempData/assets/cardBg4.png);"
      >
        <div
          class="_bg_459f4f _opacity_459f4f"
        />
        <div
          class="_layout_ecf88f"
        >
          <span
            class="_country_ecf88f"
          >
            Россия
          </span>
          <span
            class="_city_ecf88f"
          >
            Москва
          </span>
          <span
            class="_count_ecf88f"
          >
            400
            + экскурсий
          </span>
        </div>
      </div>
    </div>
    <a
      href="/locations"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-base transition-colors focus-visible:outline-none disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-grayscale-400 bg-transparent text-grayscale-400 hover:border-grayscale-500 hover:text-grayscale-500 h-14 w-72 rounded-xl"
        role="button"
      >
        перейти к списку городов
      </button>
    </a>
  </section>
</div>
`;
